neural_solution/backend/scheduler.py

```diff
@@ -0,0 +1,278 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import time
+import threading
+import subprocess
+import os
+import socket
+import re
+import json
+import glob
+import shutil
+from .task import Task
+from .cluster import Cluster
+from .task_db import TaskDB
+from neural_solution.config import INC_ENV_PATH_TEMP
+from neural_solution.backend.utils.utility import (
+    serialize,
+    dump_elapsed_time,
+    get_task_log_path,
+    get_current_time,
+    build_workspace,
+    is_remote_url,
+    get_q_model_path
+)
+from neural_solution.utils.utility import get_task_log_workspace, get_task_workspace
+from neural_solution.utils import logger
+
+cmd="echo $(conda info --base)/etc/profile.d/conda.sh"
```

Comment:
this implementation is a little surprised to me. why we have to rely on conda env?